src/Microsoft.Performance.Toolkit.Engine/EngineCreateInfo.cs

```diff
@@ -80,5 +102,23 @@ public string ExtensionDirectory
         ///     behavior.
         /// </summary>
         public VersionChecker Versioning { get; internal set; }
+
+        /// <summary>
+        ///     Gets or sets the name of the runtime on which the application is built.
+        /// </summary>
+        /// <remarks>
+        ///     Defaults to "Microsoft.Performance.Toolkit.Engine".
+        /// </remarks>
+        public string RuntimeName { get; set; } = "Microsoft.Performance.Toolkit.Engine";
+
+        /// <summary>
+        ///     Gets or sets the application name.
+        /// </summary>
+        public string ApplicationName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets a means of logging information.
+        /// </summary>
+        public ILogger Logger { get; set; }
```

Comment:
I think it would make more sense to have this be a logger factory, i.e. Func<Type, ILogger> so that the user is in control of whether the same logger gets reused across all data sources or not. As this appears to be implemented in the engine, if this property is set, then it is used for all Custom Data Sources, but if it is not set, then a new Logger is created for each Custom Data Source. Using a factory function provides that same flexibility but still allows the caller to do something else if desiered.

Reply:
Updated.